teos/src/responder.rs

```diff
@@ -366,24 +370,21 @@ impl chain::Listen for Responder {
 
         // We might be connecting a new block after a disconnection (reorg).
         // We will need to update those trackers that have been reorged.
-        let did_reorg = !self.reorged_trackers.lock().unwrap().is_empty();
-        if did_reorg {
+        if !self.reorged_trackers.lock().unwrap().is_empty() {
             // Handle reorged transactions. This clears `self.reorged_trackers`.
-            self.handle_reorged_txs(height);
+            if let Some(trackers) = self.handle_reorged_txs(height) {
+                self.gatekeeper.delete_appointments(trackers, false);
+            }
         }
 
         // Delete trackers completed at this height
-        let completed_trackers = self.check_confirmations(txs.keys().cloned().collect(), height);
-        if !completed_trackers.is_empty() {
-            self.gatekeeper
-                .delete_appointments(completed_trackers, true);
+        if let Some(trackers) = self.check_confirmations(txs.keys().cloned().collect(), height) {
+            self.gatekeeper.delete_appointments(trackers, true);
         }
 
         // Rebroadcast those transactions that need to
-        let rejected_trackers = self.rebroadcast_stale_txs(height);
-        if !rejected_trackers.is_empty() {
-            self.gatekeeper
-                .delete_appointments(rejected_trackers, false);
+        if let Some(trackers) = self.rebroadcast_stale_txs(height) {
+            self.gatekeeper.delete_appointments(trackers, false);
```

Comment:
What about defining a `mut` collection of trackers to delete and extend it inside these `if let`s so we only call `delete_appointments` once?